tests/suites/test_suite_x509parse.function

```diff
@@ -785,6 +785,69 @@ exit:
 }
 /* END_CASE */
 
+/* BEGIN_CASE depends_on:MBEDTLS_X509_CREATE_C:MBEDTLS_X509_USE_C:MBEDTLS_X509_CRT_PARSE_C:!MBEDTLS_X509_REMOVE_INFO */
+void mbedtls_x509_dn_get_next( char * name_str, int next_merged, char * expected_oids, int exp_count, char * exp_dn_gets )
+{
+    int ret = 0, i;
+    size_t len = 0;
+    mbedtls_asn1_named_data *names = NULL;
+    mbedtls_x509_name parsed, *parsed_cur, *parsed_prv;
+    unsigned char buf[1024], out[1024], *c;
```

Comment:
Do we really need stack buffers that large? The test data doesn't look that big.

Note: in the past we tended to use arbitrary large buffers in tests and not worry. Since at least a few years we try not doing that anymore, because it should be possible to run the test suites on constrained devices, where 2k on the stack might be a lot. We haven't gone and fixed all the old tests, but we try to be more careful in new tests.

Reply:
Note: it's still OK to use semi-arbitrary "round" sizes like `buf[64]` or `buf[100]` but there should be a comment relating that to the size of the expected test data, like "we only need a few dozen bytes to cover all cases, see the data file" or similar.

Reply:
There's another reason to use the exact expected size, which is that if the library overflows that size, we want the test to fail under Asan/Valgrind/Coverity. So please do use `ASSERT_ALLOC` with the exact expected size unless it's too hard to find out the expected size. At least for `out`, you do know the expected size, it's the size of the expected output, so please use that and re-allocate `out` between the two places where it's used.

Reply:
Thanks for the clarification on buffer sizing, I've updated to allocate `out` to expected size and set `buf` to the maximum required size for the test cases.